superset-frontend/src/core/sqlLab.ts

```diff
@@ -16,19 +16,175 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-import type { sqlLab as sqlLabType } from '@apache-superset/core';
+import { sqlLab as sqlLabType, core as coreType } from '@apache-superset/core';
 import {
   QUERY_FAILED,
   QUERY_SUCCESS,
   QUERY_EDITOR_SETDB,
   querySuccess,
+  startQuery,
+  START_QUERY,
+  stopQuery,
+  STOP_QUERY,
+  createQueryFailedAction,
 } from 'src/SqlLab/actions/sqlLab';
 import { RootState, store } from 'src/views/store';
 import { AnyListenerPredicate } from '@reduxjs/toolkit';
 import type { SqlLabRootState } from 'src/SqlLab/types';
 import { Disposable, Editor, Panel, Tab } from './core';
 import { createActionListener } from './utils';
 
+const { CTASMethod } = sqlLabType;
+
+export class CTAS implements sqlLabType.CTAS {
+  method: sqlLabType.CTASMethod;
+
+  tempTable: string;
+
+  constructor(asView: boolean, tempTable: string) {
+    this.method = asView ? CTASMethod.View : CTASMethod.Table;
+    this.tempTable = tempTable;
+  }
+}
+
+export class QueryContext implements sqlLabType.QueryContext {
+  clientId: string;
+
+  ctas: sqlLabType.CTAS | null;
+
+  editor: Editor;
+
+  requestedLimit: number | null;
+
+  runAsync: boolean;
+
+  startDttm: number;
+
+  tab: Tab;
+
+  private templateParams: string;
+
+  constructor(
+    clientId: string,
+    tab: Tab,
+    runAsync: boolean,
+    startDttm: number,
+    options: {
+      templateParams?: string;
+      ctasMethod?: string;
+      tempTable?: string;
+      requestedLimit?: number;
+    } = {},
+  ) {
+    this.clientId = clientId;
+    this.tab = tab;
+    this.runAsync = runAsync;
+    this.startDttm = startDttm;
+    this.requestedLimit = options.requestedLimit ?? null;
+    this.ctas = options.tempTable
+      ? new CTAS(options.ctasMethod === CTASMethod.View, options.tempTable)
+      : null;
+    this.templateParams = options.templateParams ?? '';
+  }
+
+  get templateParameters() {
```

Comment:
It would be good to add a comment explaining that we want to do the parsing only when the property is accessed. That's why we didn't add to the constructor.